src/modules/network.js

```diff
@@ -0,0 +1,114 @@
+import List from './lists.js';
+import Popup from './popup.js';
+
+export default class {
+  constructor(menuElement, listElement) {
+    this.meals = 'https://themealdb.com/api/json/v1/1/';
+    this.involvement = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/';
+    this.appId = '7f1BuLlilpO9a0A6DJ97';
+    this.list = new List(menuElement, listElement);
+    this.popup = new Popup();
+  }
+
+  buildHeaders = (method, body) => ({ method, body: JSON.stringify(body), headers: { 'content-type': 'application/json' } });
+
+  buildRequest = async (url, options = {}) => {
+    const response = await fetch(url, options);
+    const result = await response.text();
+    if (result.length > 0 && Object.keys(options).length === 0) {
+      return JSON.parse(result);
+    }
+    return result;
+  }
+
+  sendRequest = async (action, promises, item = null) => {
+    const responses = await Promise.all(promises);
+    this.handleResponse(action, responses, item);
+  }
+
+  handleResponse = (action, responses, item) => {
+    switch (action) {
+      case 'getList':
+        this.list.displayList(
+          {
+            category: item, listItems: responses[0].meals, totalLikes: responses[1],
+          },
+        );
+        break;
+      case 'postLike':
+        this.list.updateLikes(item);
+        break;
+      case 'getComments':
+        this.popup.renderPopup({ popupType: 'comments', itemDetails: responses[0], involvementList: responses[1] });
```

Comment:
The property name used for determining `popupType` in` Popup Class` is called `targetType`. Make sure to refactor this in order to avoid undefined object item name while trying to access it

Reply:
Great observation 👏🏻 I will change it 👌🏻

Reply:
Good job